### ISSUE TITLE
Implementation questions

### ISSUE THREAD

**Issue by leonardohn** [KindXiaoming/grow-crystals#3]
First of all, congratulations for the amazing work.

I have a few questions about implementation decisions related to the MNIST example. Why is the harmonic loss a part of the network architecture instead of being a separate class like CrossEntropyLoss? Also, why is CrossEntropyLoss being used as the optimizer's loss function when harmonic=True?

Thank you in advance.

**Comment by leonardohn** [comment]
My bad, I totally missed that the criterion variable was dangling in the first experiment. As for the first question, as it is really close to issue #2, I'm marking this as a duplicate.
